lib/net.c

```diff
@@ -43,16 +45,43 @@
 static struct net_config config;
 
 /*
- * XXX The secret key of the RSS hash must be random
- * in order to avoid hackers to know it.
+ * The secret key of the RSS hash (RSK) must be random in order
+ * to prevent hackers from knowing it.
+ * To validate if the key generated is reasonable, the
+ * number of bits set to 1 in the key must be greater than 
+ * 10% and less than 90% of the total bits in the key.
+ * min_num_set_bits and max_num_set_bits represent the lower 
+ * and upper bound for the key.
  */
-uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN] = {
-	0x6d, 0x5a, 0x56, 0xda, 0x25, 0x5b, 0x0e, 0xc2,
-	0x41, 0x67, 0x25, 0x3d, 0x43, 0xa3, 0x8f, 0xb0,
-	0xd0, 0xca, 0x2b, 0xcb, 0xae, 0x7b, 0x30, 0xb4,
-	0x77, 0xcb, 0x2d, 0xa3, 0x80, 0x30, 0xf2, 0x0c,
-	0x6a, 0x42, 0xb7, 0x3b, 0xbe, 0xac, 0x01, 0xfa,
-};
+uint8_t default_rss_key[GATEKEEPER_RSS_KEY_LEN];
+
+static void
+randomize_rss_key(void)
+{
+	uint8_t i;
+	uint8_t final_set_count = 0;
+	ssize_t number_of_bytes;
+	uint16_t min_num_set_bits = sizeof(default_rss_key) * 8 * 0.1;
+	uint16_t max_num_set_bits = sizeof(default_rss_key) * 8 * 0.9;
+
+	while (final_set_count < min_num_set_bits ||
+			final_set_count > max_num_set_bits) {
+
+		/* 
+		 * The last parameter to syscall getrandom i.e flags will alter the 
+		 * behaviour of the call. In the case where flag == 0, the call will
+		 * block until the pool has the required entropy.  The GRND_RANDOM flag
+		 * bit can be used to switch to the /dev/random pool.
+		 */
+		while ((number_of_bytes = syscall(SYS_getrandom, default_rss_key, sizeof(default_rss_key), 0)) !=
```

Comment:
Coding style: please make sure that each line of code should not exceed 80 characters.

Reply:
Shouldn't we use ``GRND_RANDOM`` instead of ``0`` in the ``flag`` parameter? Since we are trying to ensure the randomness as much as possible?

Reply:
A `0` in the flag parameter means that the `/dev/urandom` will be used for generating the random number. The `/dev/random` which is used if the `GRND_RANDOM` is passed to the flags. It is recommended to use `/dev/urandom` for most practical purposes. More information here https://unix.stackexchange.com/questions/324209/when-to-use-dev-random-vs-dev-urandom#324210

Reply:
This is the rule I referred from DPDK's coding style page :
> If you have to wrap a long statement, put the operator at the end of the line, and indent again.

Reply:
See the comment below.

Reply:
For the flag parameter, could you please add more comments on the choice of ``0`` explaining the reasons? The materials you pointed to are quite helpful.